react-components/src/data-providers/core-dm-provider/filterNodesByMappedTo3d.ts

```diff
@@ -0,0 +1,251 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+import { type InstancesWithView } from '../../query/useSearchMappedEquipmentFDM';
+import {
+  type QueryRequest,
+  type QueryTableExpressionV3,
+  type SourceSelectorV3
+} from '@cognite/sdk';
+import { getDirectRelationProperties } from '../utils/getDirectRelationProperties';
+import {
+  type Cognite3DObjectProperties,
+  type COGNITE_3D_OBJECT_SOURCE,
+  COGNITE_CAD_NODE_SOURCE,
+  COGNITE_POINT_CLOUD_VOLUME_SOURCE,
+  COGNITE_VISUALIZABLE_SOURCE,
+  CORE_DM_3D_CONTAINER_SPACE
+} from './dataModels';
+import { type DmsUniqueIdentifier, type FdmSDK } from '../FdmSDK';
+import { cogniteObject3dSourceWithProperties } from './cogniteObject3dSourceWithProperties';
+import { type FdmKey } from '../../components/CacheProvider/types';
+import { createFdmKey } from '../../components/CacheProvider/idAndKeyTranslation';
+import { type PromiseType } from '../utils/typeUtils';
+import { isString } from 'lodash';
+import { type QueryResult } from '../utils/queryNodesAndEdges';
+
+export async function filterNodesByMappedTo3d(
+  nodes: InstancesWithView[],
+  revisionRefs: DmsUniqueIdentifier[],
+  _spacesToSearch: string[],
+  fdmSdk: FdmSDK
+): Promise<InstancesWithView[]> {
+  const connectionData = await fetchConnectionData(nodes, revisionRefs, fdmSdk);
+
+  const object3dKeys: Set<FdmKey> = createRelevantObject3dKeys(connectionData);
+
+  const result = nodes.map((viewWithNodes) => {
+    if (viewWithNodes.view.externalId !== 'CogniteAsset') {
+      return {
+        view: viewWithNodes.view,
+        instances: []
+      };
+    }
+    return {
+      view: viewWithNodes.view,
+      instances: viewWithNodes.instances.filter((instance) => {
+        const object3dId = instance.properties.object3D as unknown as
+          | DmsUniqueIdentifier
+          | undefined;
+        if (!isString(object3dId?.externalId) || !isString(object3dId?.space)) {
+          return false;
+        }
+        return object3dKeys.has(createFdmKey(object3dId));
+      })
+    };
+  });
+
+  return result;
+}
+
+function createRelevantObject3dKeys(
+  connectionData: PromiseType<ReturnType<typeof fetchConnectionData>>
+): Set<FdmKey> {
+  const cadObject3dList = [...connectionData.items.initial_nodes_cad_nodes]
+    .concat(connectionData.items.direct_nodes_cad_nodes)
+    .concat(connectionData.items.indirect_nodes_cad_nodes)
+    .map((node) => createFdmKey(node.properties.cdf_cdm['CogniteCADNode/v1'].object3D));
+
+  const pointCloudObject3dList = [...connectionData.items.initial_nodes_point_cloud_volumes]
+    .concat(connectionData.items.direct_nodes_point_cloud_volumes)
+    .concat(connectionData.items.indirect_nodes_point_cloud_volumes)
+    .map((pointCloudVolume) =>
+      createFdmKey(pointCloudVolume.properties.cdf_cdm['CognitePointCloudVolume/v1'].object3D)
+    );
```

Comment:
Right, I'll do it :+1: I didn't think about it, because it's auto-completed as I access the properties, but I agree that it's easier to maintain if we factor them out as constants :+1: